### PR TITLE
Hotfix - Fix Object Resolver Instantiating Product Class With Wrong Signature

### PR DESCRIPTION
**Pull request by supercid**

### CODE REVIEW

File: classes/helpers/NostoHelperController.php [hunk 1]
```diff
@@ -75,12 +75,17 @@ public static function resolveObject($idName, $klass, $method, $multiLanguageObj
...
-        if ($object instanceof $klass == false) {
-            $id = null;
-            if (Tools::getValue($idName)) {
-                $id = Tools::getValue($idName);
-            }
-            if ($id) {
+        $id = Tools::getValue($idName) ?: null;
+        if ($object instanceof $klass === false && $id !== null) {
+            if ($klass === 'Product') {
+                $object = new $klass
```
Comment (chenliangde) [inline classes/helpers/NostoHelperController.php]: Is this constructor signature same in all prestashop version?

Comment (supercid) [reply classes/helpers/NostoHelperController.php]: Yes, even before 1.5.5:
https://github.com/PrestaShop/PrestaShop/blame/1.5.5.0/classes/Product.php#L443

https://github.com/PrestaShop/PrestaShop/blame/develop/classes/Product.php#L548

Comment (hannunosto) [reply classes/helpers/NostoHelperController.php]: How about using Reflection here to figure out the correct parameters for `__constructor`? The `shopId` is also passed to all constructors but for example `Manufacturer` and `Order` only take `id` and `id_lang`. 

Here's something to start with
```
$reflectionMethod = new ReflectionMethod($klass, '__construct');
$constructorArgs = $reflectionMethod->getParameters();
$argsForObject = array();
$argsForObject[] = $id;
// Check here what arguments needs to be passed 
$reflectionClass = new ReflectionClass('Foo');
$object = $reflectionClass->newInstanceArgs($argsForObject);
```

Comment (supercid) [reply classes/helpers/NostoHelperController.php]: Great idea

**Pull request merged**